p2p/pkg/rpc/bidder/service.go

```diff
@@ -131,17 +140,42 @@ func (s *Service) SendBid(
 		}
 		return stripped
 	}
-	txnsStr := strings.Join(stripPrefix(bid.TxHashes), ",")
-	revertingTxHashesStr := strings.Join(stripPrefix(bid.RevertingTxHashes), ",")
+	var (
+		txnsStr          string
+		revertingTxnsStr string = strings.Join(stripPrefix(bid.RevertingTxHashes), ",")
+	)
+	switch {
+	case len(bid.TxHashes) > 0:
+		txnsStr = strings.Join(stripPrefix(bid.TxHashes), ",")
+	case len(bid.RawTransactions) > 0:
+		for _, rawTx := range bid.RawTransactions {
+			rawTxnBytes, err := hex.DecodeString(strings.TrimPrefix(rawTx, "0x"))
+			if err != nil {
+				s.logger.Error("decoding raw transaction", "error", err)
+				return status.Errorf(codes.InvalidArgument, "decoding raw transaction: %v", err)
+			}
+			txnObj := new(types.Transaction)
+			err = txnObj.UnmarshalBinary(rawTxnBytes)
+			if err != nil {
+				s.logger.Error("unmarshaling raw transaction", "error", err)
+				return status.Errorf(codes.InvalidArgument, "unmarshaling raw transaction: %v", err)
+			}
+			txnsStr += strings.TrimPrefix(txnObj.Hash().Hex(), "0x") + ","
+		}
+		txnsStr = strings.TrimSuffix(txnsStr, ",")
+	}
 
 	respC, err := s.sender.SendBid(
 		ctx,
-		txnsStr,
-		bid.Amount,
-		bid.BlockNumber,
-		bid.DecayStartTimestamp,
-		bid.DecayEndTimestamp,
-		revertingTxHashesStr,
+		&preconfirmationv1.Bid{
+			TxHash:              txnsStr,
+			BidAmount:           bid.Amount,
+			BlockNumber:         bid.BlockNumber,
+			DecayStartTimestamp: bid.DecayStartTimestamp,
+			DecayEndTimestamp:   bid.DecayEndTimestamp,
+			RevertingTxHashes:   revertingTxnsStr,
```

Comment:
We could skip var creation and do initialization here:
```suggestion
			RevertingTxHashes:   strings.Join(stripPrefix(bid.RevertingTxHashes), ","),
```